### ISSUE TITLE
Product Gallery: Add keyboard navigation

### ISSUE THREAD

**Issue by danieldudzic** [woocommerce/woocommerce-gutenberg-products-block#11284]
The **Product Gallery** block needs to support keyboard navigation (and should be able to pass an accessibility audit).

**Comment by roykho** [comment]
To add to this, it would be good to also add the `ESC` key where when pressed would close the overlay lightbox/dialog.
